debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/UnwrapFromMongoDbEnvelope.java

```diff
@@ -139,8 +201,38 @@ public R apply(R r) {
 
             // update
             if (patchRecord.value() != null) {
-                valueDocument = BsonDocument.parse(patchRecord.value().toString());
-                valueDocument = valueDocument.getDocument("$set");
+                document = BsonDocument.parse(patchRecord.value().toString());
+
+                valueDocument = new BsonDocument();
+
+                if (document.containsKey("$set")) {
+                    valueDocument = document.getDocument("$set");
+                }
+
+                if (document.containsKey("$unset")) {
+                    Set<Entry<String, BsonValue>> unsetDocumentEntry = document.getDocument("$unset").entrySet();
+
+                    for (Entry<String, BsonValue> valueEntry : unsetDocumentEntry) {
+                        // In case unset of a key is false we don't have to do anything with it,
```

Comment:
This one is still a bit blurry to me: I thought whatever value is passed for an unset field is ignored by MongoDB? I think we were discussing it before, but I don't quite remember the outcome. Thanks for any pointer!

Reply:
Sure thing, some context:
https://github.com/debezium/debezium/pull/669#issuecomment-437889390
https://github.com/debezium/debezium/pull/669#issuecomment-439340471

So basically, there's no proof it can be false, but also I imagine it can never be produced in the source code, in this case I think the current solution is safe and more explicit

Reply:
Ah yes, gotcha. Thanks!